src/openms/source/KERNEL/MassTrace.cpp

```diff
@@ -283,12 +283,12 @@ namespace OpenMS
     return fwhm_;
   }
 
-    double MassTrace::calculateXCoordinateAtHalfMax_(double xA, double xB, double yA, double yB, double fwhm_int) const
+    double MassTrace::linearInterpolationAtY_(double xA, double xB, double yA, double yB, double y_eval) const
     {
       // no solution -> return an estimate
       if (std::fabs(xA - xB) == 0 || std::fabs(yA - yB) == 0)  { return xA; }
 
-      double xC = (xA + ((fwhm_int - yA) * (xB - xA) / (yB - yA)));
+      double xC = (xA + ((y_eval - yA) * (xB - xA) / (yB - yA)));
```

Comment:
can you add a `POSTCONDITION` to check that `xC` is between `xA` and `xB` to make sure we are interpolating not extrapolating?

Reply:
Sure.